docs/chapter-14.rst

```diff
@@ -2,102 +2,203 @@
 Grid
 ====
 
-py4web comes with a Grid object providing grid and CRUD capabilities.
+py4web comes with a Grid object providing grid and CRUD (create, update and delete) capabilities.
+This allows you to quickly and safely provide an interface to your data. Since it's also
+highly customizable, it's the corner stone of most py4web's applications.
 
 Key Features
 ------------
 
+-  Full CRUD with Delete Confirmation
 -  Click column heads for sorting - click again for DESC
 -  Pagination control
 -  Built in Search (can use search_queries OR search_form)
 -  Action Buttons - with or without text
--  Full CRUD with Delete Confirmation
 -  Pre and Post Action (add your own buttons to each row)
 -  Grid dates in local format
 -  Default formatting by type plus user overrides
 
 Basic Example
 -------------
 
-In this simple example we will make a grid over the company table.
+In this simple example we will make a grid over the superhero table.
 
-controllers.py
+Create a new minimal app called ``grid``. Change it with the following content.
 
 .. code:: python
 
-   from functools import reduce
+   # in grid/__init__.py
+   import os
    from py4web.utils.grid import Grid
-   from py4web import action
-   from .common import db, session, auth
+   from py4web import action, Field, DAL
+
+
+   # database definition
+   DB_FOLDER = os.path.join(os.path.dirname(__file__), 'databases')
+   if not os.path.isdir(DB_FOLDER):
+      os.mkdir(DB_FOLDER)
+   db = DAL('sqlite://storage.sqlite', folder=DB_FOLDER)
+   db.define_table(
+      'person',
+      Field('name'),
+      Field('job'))
+
+   # add example entries in db
+   if not db(db.person).count():
+      db.person.insert(name='Clark Kent', job='Journalist')
+      db.person.insert(name='Peter Park', job='Photographer')
+      db.person.insert(name='Bruce Wayne', job='CEO')
+      db.commit()
+
+   @action('index', method=['POST', 'GET'])
+   @action('index/<path:path>', method=['POST', 'GET'])
+   @action.uses(db, 'grid.html')
+   def index(path=None):
+         grid = Grid(path,
+                  formstyle=FormStyleDefault, # FormStyleDefault or FormStyleBulma
+                  grid_class_style=GridClassStyle, #GridClassStyle or GridClassStyleBulma      
```

Comment:
If this example is supposed to be able to be copy pasted, these two classes should be imported at the top. If it's just a generic example, then this is fine.

Reply:
You're right, thank you! fixed